### PR TITLE
add camera.set_position to TrackballCamera

### PR DESCRIPTION
**Pull request by chrishavlin**
This adds a camera.set_position function so that the view_matrix will get updated when setting a position.

here's the result of the additional image i added to examples/amr_osmesa.py

step3_set_position

### CODE REVIEW

File: yt_idv/cameras/trackball_camera.py [hunk 1]
```diff
@@ -86,3 +86,7 @@ def offset_position(self, dpos=None):
 
     def _compute_matrices(self):
         pass
+
+    def set_position(self, pos):
+        self.position = pos
+        self.view_matrix = get_lookat_matrix(self.position, self.focus, self.up)
```
Comment (chrishavlin) [inline yt_idv/cameras/trackball_camera.py:92]: wait, i should just call `_update_matrices` here I think. will update.

**Comment by chrishavlin** [comment]
drafting because looking at the base camera, looks like there was already some architecture for this, just not implemented fully? 

https://github.com/yt-project/yt_idv/blob/e093d4e91283c484e42277d749221e86649d2478/yt_idv/cameras/base_camera.py#L74-L87

will see if it's a quick thing, cause i'd prefer to do thsi with traitlets.

**Comment by chrishavlin** [comment]
ya, that makes sense. i was getting confused about the possible dependencies and order of operations -- e.g., if you update the orientation in the trackball you need to re-calculate the position, but if you set the position you would need to update the orientation (and other matrices). so having a single state object of some kind would make a lot of sense. 

you think it's OK to merge this one in the meantime though? It'd help with the tests I'm adding in #159

**Comment by matthewturk** [comment]
I trust your judgment, yup!

**Pull request merged**